### PR TITLE
[usbdev] Rewrite usbdev resume link active seq description

### PR DESCRIPTION
**Pull request by Saad-Anees1**
This PR includes the modification to the description of the usbdev resume_link_active sequence.

### CODE REVIEW

**Comment by Saad22386** [comment]
@rswarbrick

**Comment by rswarbrick** [comment]
@alees24: Since you know rather more about USB than I do... Would you mind taking a look at this and checking it makes sense? (I suspect it does, but want to get an experienced pair of eyes!)

File: hw/ip/usbdev/data/usbdev_testplan.hjson [hunk 1]
```diff
@@ -431,8 +431,8 @@
             then write will only have an effect when the device is in the LinkPowered state.
 
             - Perform no operation in J state for more than 3 ms.
-            - Read the link state from usbstate register to see if it is in powered state and then send the
-              resume signal (K state) for 20 ms and set the usbctrl.resume_link_active.
+            - Read the link state from usbstate register to see if it is in powered state and set the
+              usbctrl.resume_link_active.
             - Verify that the resume_link_active interrupt goes high and device will be resumed.
```
Comment (alees24) [inline hw/ip/usbdev/data/usbdev_testplan.hjson:436]: The description change is okay but the register name is 'usbstat' without a trailing 'e.'

Also, whilst this test description is being changed, please note that there is no 'resume_link_active interrupt' and the test should instead be looking at the link state in 'usbstat' to ascertain whether the 'resuming' state has been entered.

**Review by alees24 — CHANGES_REQUESTED** [review]

File: hw/ip/usbdev/data/usbdev_testplan.hjson (hunk 1 above)
Comment (Saad22386) [reply hw/ip/usbdev/data/usbdev_testplan.hjson:436]: > The description change is okay but the register name is 'usbstat' without a trailing 'e.'
> 
> Also, whilst this test description is being changed, please note that there is no 'resume_link_active interrupt' and the test should instead be looking at the link state in 'usbstat' to ascertain whether the 'resuming' state has been entered.

Are you suggesting that the "link_resume" interrupt in the intr_state register doesn't monitor the resuming state?

**Comment by Saad22386** [comment]
When I set usbctrl.resume_link_active to 1 after the usbdev is in the powered state, it randomly transitions to either the active_nosof state or the resuming state. However, according to the usbctrl.resume_link_active bit description, it should be in the resuming state. Why does this occur?

**Comment by alees24** [comment]
> When I set usbctrl.resume_link_active to 1 after the usbdev is in the powered state, it randomly transitions to either the active_nosof state or the resuming state. However, according to the usbctrl.resume_link_active bit description, it should be in the resuming state. Why does this occur?

I suspect that the usb20_driver is interfering with the test by driving a Bus Reset or Idle signal into the USB device.
The 'resume_link_active' bit is supposed to be set by software when the USB device is powered up whilst the usbdev_aon_wake module still has control of the bus. Software sets this bit and then enables the pull up, which causes the link state to transition to Powered and then into Resuming.

During this 'Resume Signaling' period (7.1.7.7 in the spec) there should be at least 20ms of K state, but the usb20_driver is presently unaware of suspend/resume behavior and it automatically drives a Bus Reset and Idle period onto the USB when activated.

**Comment by alees24** [comment]
Thanks for this PR. Superseded now that USBDEV has reached V2S status.

**Pull request closed without merging**